utils/tinystr/src/int_ops.rs

```diff
@@ -98,6 +98,43 @@ impl Aligned4 {
         (invalid_case & 0x8080_8080) == 0x8080_8080
     }
 
+    pub const fn is_ascii_alphabetic_lowercase(&self) -> bool {
+        let word = self.0;
+        // See explanatory comments in is_ascii_alphabetic
+        let mask = (word + 0x7f7f_7f7f) & 0x8080_8080;
+        let lower = word | 0x2020_2020;
+        let alpha = !(lower + 0x1f1f_1f1f) | (lower + 0x0505_0505);
+        // See explanatory comments in is_ascii_lowercase
+        let invalid_case = !(word + 0x3f3f_3f3f) | (word + 0x2525_2525);
+        (alpha & mask) ^ (invalid_case & 0x8080_8080) == 0x8080_8080
```

Comment:
Suggestion: All we need to do is find characters that are loweralpha. All others are invalid.

The following should work:

```suggestion
        let mask = (word + 0x7f7f_7f7f) & 0x8080_8080;
        let lower_alpha = !(word + 0x1f1f_1f1f) | (word + 0x0505_0505);
        (lower_alpha & mask) == 0
```

Similar changes can be made in the other functions.

Reply:
That makes perfect sense -- thank you for the suggestion!